pkgs/top-level/all-packages.nix

```diff
@@ -3315,7 +3315,12 @@ with pkgs;
 
   astc-encoder = callPackage ../tools/graphics/astc-encoder { };
 
-  asymptote = libsForQt5.callPackage ../tools/graphics/asymptote { };
+  asymptote = libsForQt5.callPackage ../tools/graphics/asymptote {
+    stdenv = if stdenv.isDarwin
+      # SDK >= 10.13 required for open_memstream
```

Comment:
Have you tried adding `memstreamHook` to buildInputs?

Reply:
I think memstream should compile to shared library (dylib) for this to work... ? I have even tried setting `LDFLAGS`, but it still doesn't pick it up.

Reply:
It should set ldflags for you https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/memstream/setup-hook.sh

Reply:
Somehow the linker ignores `libmemstream.a`. I tried the manual `LDFLAGS` to see `-L/.../memstream-0.1/lib -lmemstream` on the command line, but it didn't make a difference. I am not sure what is going on here.

Reply:
To avoid holding TeX Live back, I suggest I drop the V3D/XDR support for Darwin temporarily (right now it's missing anyway, and nobody complained). I can open a new issue/PR to find the best way to reenable it. Does that sound good?

Reply:
I don't mind if we keep your current version with use of newer SDK. I appreciate you trying this out.

Reply:
> > use of newer SDK
> 
> I feel a little guilty breaking TeX Live on old macs!

I don't think that's the case. My concern was more about possibly leaking and mixing different stdenv's (probably a rare case).